### PR TITLE
fix(Jenkinsfile): do timeout for each step on it's own

### PR DESCRIPTION
**Pull request by fruch**
since a global timeout of the whole stage of provision tests can leave some of the resources not cleaned, and also the logs uncollected

Fixes: #6875

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by vponomaryov** [comment]
> @vponomaryov can you take a look at the k8s provision tests, as to why are the failing ?

After the `test_drain_kubernetes_node_then_wait_and_replace_scylla_node` test that passed we get following failure processing the `db_cluster` fixture:
```
< t:2023-11-30 10:14:04,080 f:base.py         l:142  c:KubernetesCmdRunner  p:DEBUG > Command "/usr/bin/cqlsh --no-color   --request-timeout=120 --connect-timeout=60  -e "describe keyspaces" 10.16.218.207 9042" finished with status 0
< t:2023-11-30 10:14:04,110 f:remote_base.py  l:521  c:KubernetesCmdRunner  p:DEBUG > Running command "/usr/bin/cqlsh --no-color   --request-timeout=120 --connect-timeout=60  -e "consistency QUORUM;SELECT COUNT(*) FROM keyspace1.standard1" 10.16.218.207 9042"...
< t:2023-11-30 10:14:04,716 f:base.py         l:222  c:KubernetesCmdRunner  p:DEBUG > <stdin>:1:NoHostAvailable: 
< t:2023-11-30 10:14:04,716 f:base.py         l:222  c:KubernetesCmdRunner  p:DEBUG > Consistency level set to QUORUM.
< t:2023-11-30 10:14:04,792 f:base.py         l:146  c:KubernetesCmdRunner  p:ERROR > Error executing command: "/usr/bin/cqlsh --no-color   --request-timeout=120 --connect-timeout=60  -e "consistency QUORUM;SELECT COUNT(*) FROM keyspace1.standard1" 10.16.218.207 9042"; Exit status: 2
< t:2023-11-30 10:14:04,792 f:base.py         l:148  c:KubernetesCmdRunner  p:DEBUG > STDOUT: Consistency level set to QUORUM.

< t:2023-11-30 10:14:04,792 f:base.py         l:150  c:KubernetesCmdRunner  p:DEBUG > STDERR: <stdin>:1:NoHostAvailable: 
```

And it happens for each of the following tests.
Strange that first `cqlsh` calls works - `describe keyspaces`,
but following `consistency QUORUM;SELECT COUNT(*) FROM keyspace1.standard1` fails with `NoHostAvailable`.

File: Jenkinsfile [hunk 1]
```diff
@@ -230,10 +229,12 @@ pipeline {
                                         try {
                                             wrap([$class: 'BuildUser']) {
                                                 env.BUILD_USER_ID=env.CHANGE_AUTHOR
-                                                dir(working_dir) {
-                                                    runSctTest(curr_params, builder.region, curr_params.get('functional_tests', false))
-                                                    result = 'SUCCESS'
-                                                    pullRequestSetResult('success', "jenkins/provision_${backend}", 'All test cases are passed')
+                                                timeout(time: 180, unit: 'MINUTES') {
```
Comment (vponomaryov) [inline Jenkinsfile:232]: `3h` is not enough for the k8s functional tests.

Here is how much take took K8S functional tests without SNI ones: `27 passed, 7 skipped in 11103.47s (3:05:03)`

So, probably we should set `5h` here with idea of future tests addition.

**Comment by vponomaryov** [comment]
Default Scylla versions update: https://github.com/scylladb/scylla-cluster-tests/pull/6883

File: Jenkinsfile (hunk 1 above)
Comment (fruch) [reply Jenkinsfile:232]: we probably want the SNI disabled by default, given what was discussed in the weekly call

Comment (vponomaryov) [reply Jenkinsfile]: I don't mind to change it for PR checks while we have weekly runs for both configurations.

Comment (fruch) [reply Jenkinsfile]: also for the weekly, we might disable it, until they resurrect it.

Comment (vponomaryov) [reply Jenkinsfile]: https://github.com/scylladb/scylla-cluster-tests/pull/6903

**Pull request merged**